autowarefoundation/autoware
# Feature/enet image segmenter

## Status
**DEVELOPMENT**

## Description
This PR adds image segmentation capabilities to Autoware using ENet segmentation network.

## Related Issue
Defined in autowarefoundation/autoware_ai#968

## Todos

## Steps to Test or Reproduce
1. Follow the README file in the ENet PAckage to install, compile the corresponding Caffe Fork and download pretrained models.
https://github.com/CPFL/Autoware/blob/feature/enet_image_segmenter/ros/src/computing/perception/detection/packages/enet_image_segmenter/README.md

2. Play a ROSBag containing image data, make sure it's being published in `/image_raw` or change the launch file accordingly or relay.

3. Launch the node
```
roslaunch image_segmenter image_segmenter_enet.launch
```
The node will subscribe to `/image_raw`, it will publish `/image_segmented` and `/image_segmented_blended` as shown in:
https://drive.google.com/open?id=0BzYuVrO9pnh6MmNHVjJZSzZnUGc

**Successfuly compiled on Ubuntu 14.04 and 16.04**
